sources/ClangSharp/Cursors/Cursor.cs

```diff
@@ -80,6 +80,10 @@ internal static Cursor Create(CXCursor handle)
             {
                 result = Attr.Create(handle);
             }
+            else if (handle.IsPreprocessing)
+            {
+                result = Preprocessing.Create(handle);
```

Comment:
nit: I've tried to mirror the Clang C++ API structure where applicable.

In this case, it would be:
```csharp
class PreprocessedEntity : Cursor  { }
class MacroExpansion : PreprocessedEntity { }
class PreprocessingDirective : PreprocessedEntity { }
class InclusionDirective : PreprocessingDirective { }
class MacroDefinitionRecord : PreprocessingDirective { }
```

Reply:
Following the C++ API gives a structured layout that closely mirrors the actual OOP API design and allows various bits of code to be shared where it makes sense due to the backing data the cursor handles represent